homeassistant/components/mawaqit/sensor.py

```diff
@@ -0,0 +1,203 @@
+
+    client = hass.data[DOMAIN]
+    if not client:
+        _LOGGER.error("Error retrieving client object")
+
+    entities = []
+    for sensor_type in SENSOR_TYPES:
+        if sensor_type in [
+            "Fajr",
+            "Shurouq",
+            "Dhuhr",
+            "Asr",
+            "Maghrib",
+            "Isha",
+            "Jumua",
+            "Jumua 2",  # "Aid" and "Aid 2",
+            "Fajr Iqama",
+            "Shurouq Iqama",
+            "Dhuhr Iqama",
+            "Asr Iqama",
+            "Maghrib Iqama",
+            "Isha Iqama",
+            "Next Salat Name",
+            "Next Salat Time",
+            "Next Salat Preparation",
+        ]:
+            sensor = MawaqitPrayerTimeSensor(sensor_type, client)
+            entities.append(sensor)
+    async_add_entities(entities, True)
+
+    name = "My Mosque"
+    sensor1 = [MyMosqueSensor(name, hass)]
+    async_add_entities(sensor1, True)
+
+
+class MawaqitPrayerTimeSensor(SensorEntity):
+    """Representation of an Mawaqit prayer time sensor."""
+
+    def __init__(self, sensor_type, client) -> None:
+        """Initialize the Mawaqit prayer time sensor."""
+        self.sensor_type = sensor_type
+        self.client = client
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return f"{self.sensor_type} {SENSOR_TYPES[self.sensor_type]}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique id of the entity."""
+        return self.sensor_type
+
+    @property
+    def icon(self) -> str:
+        """Icon to display in the front end."""
+        return PRAYER_TIMES_ICON
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor.  .astimezone(dt_util.UTC)."""
+        if self.sensor_type in [
+            "Fajr",
+            "Shurouq",
+            "Dhuhr",
+            "Asr",
+            "Maghrib",
+            "Isha",
+            "Jumua",
+            "Jumua 2",  # "Aid", "Aid 2",
+            "Fajr Iqama",
+            "Shurouq Iqama",
+            "Dhuhr Iqama",
+            "Asr Iqama",
+            "Maghrib Iqama",
+            "Isha Iqama",
+            "Next Salat Time",
+            "Next Salat Preparation",
+        ]:
+            time = self.client.prayer_times_info.get(self.sensor_type)
+            _LOGGER.debug("[;] before %s Time: %s", self.sensor_type, time)
+            if time is not None:
+                _LOGGER.debug(
+                    "[;] %s Time: %s", self.sensor_type, time.astimezone(dt_util.UTC)
+                )
+                return time.astimezone(dt_util.UTC)
+
+            return None
+
+        return self.client.prayer_times_info.get(self.sensor_type)
+
+    @property
+    def should_poll(self) -> bool:
+        """Disable polling."""
+        return False
+
+    @property
+    def device_class(self) -> SensorDeviceClass | None:
+        """Return the device class."""
+        if self.sensor_type in [
```

Comment:
Instead of having this if statements, please check out the other integration and use similar to them entity descriptions